tests/harness/docs/MemoryLoopbackExpanded.cpp

```diff
@@ -0,0 +1,589 @@
+#include <core/simulator_layout.hpp>
```

Comment:
I can explain a bit about how the preprocessing happens in compilers. As that's what the command does. I'll explain the usage of each flag. Also the reason we get such a large file is that g++ expands the headers recursively. I'll describe what to look for in this large file.
Would this be sufficient?

Reply:
I've added all this and a doodle as well. The doodle is ugly though. What do you think?

Reply:
I looked at the docs and doodle. It's correct but not really what I was looking for, because it explains something very basic (essentially, how the preprocessor handles #includes).

What I think would be very high value here is an explanation of the various boost macros that are in use, and how the MemoryLoopbackImpl file gets expanded/elaborated with the assistance of the various macros. This is something which I think only the original Flexus architects knew - now that you have spent a summer working on this, it would be good to write some of that down.

Reply:
Sure. I'll add some sentences on that as well

Reply:
I've added brief descriptions for each macro. Please have a look.